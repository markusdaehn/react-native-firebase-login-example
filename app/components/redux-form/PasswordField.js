
import React, { Component } from 'react';
import { Field } from 'redux-form';
import Input from './Input';

export default class PasswordField extends Component {
  constructor(props) {
    super(props);
  }

  focus() {
    this.field.getRenderedComponent().focus();
  }

  render() {
    const { name, placeholder, ...otherProps } = this.props;
    console.log(`typeof(name) = ${name}`);
    return (
      <Field
        name={name || 'password'}
        ref={(field) => this.field = field}
        withRef
        placeholder={placeholder || 'Enter a password'}
        secureTextEntry
        component={Input}
        {...otherProps}
      />
    );
  }
}
