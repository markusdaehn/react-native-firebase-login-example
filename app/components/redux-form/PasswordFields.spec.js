import 'react-native';
import React from 'react';
import PasswordField from './PasswordField';
import { shallow } from 'enzyme';

describe('PasswordField', function () {
  let password;
  describe('when rendered with no props', function () {
    beforeEach(function () {
      password = shallow(<PasswordField />)
    });

    it('should render correctly with default props', function () {
      expect(password).toMatchSnapshot();
    });
  });
  describe('when rendered with name and placeholder properties set', function () {
    beforeEach(function () {
      password = shallow(<PasswordField name='password-field-name' placeholder='password-field-placeholder'  />)
    });
    it('should render correctly with name and placeholder set to passed in props', function () {
      expect(password).toMatchSnapshot();
    });
  });
});
