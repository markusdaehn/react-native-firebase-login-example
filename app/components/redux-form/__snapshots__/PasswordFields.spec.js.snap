// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordField when rendered with name and placeholder properties set should render correctly with name and placeholder set to passed in props 1`] = `
<Field
  component={[Function]}
  name="password-field-name"
  placeholder="password-field-placeholder"
  secureTextEntry={true}
  withRef={true}
/>
`;

exports[`PasswordField when rendered with no props should render correctly with default props 1`] = `
<Field
  component={[Function]}
  name="password"
  placeholder="Enter a password"
  secureTextEntry={true}
  withRef={true}
/>
`;
